### PR TITLE
[FEATURE][ML] Change df-analytics config to accept single analysis

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
This changes the config to have a single analysis using the names writeables
framework. `analyses_fields` are also renamed to `analyzed_fields`.

Finally, during this work it was revealed that named writeables should not
be declared from the `MachineLearning` plugin but in the `XPackClientPlugin`
instead, so this is also done for the evaluation named writeables. Note that
once the transport client is removed they will move back into the `MachineLearning`
plugin.

### CODE REVIEW

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/packaging-sample

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/2

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/DataFrameAnalysis.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.dataframe.analyses;
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/DataFrameAnalysis.java]: This package is indeed supposed to house all analysis we'll implement.

**Pull request merged**